src/server/controllers/training.controller.ts

```diff
@@ -0,0 +1,40 @@
+import { GetByIdInput } from '~/server/schema/base.schema';
+import { getModel } from '~/server/services/model.service';
+import { throwDbError } from '~/server/utils/errorHandling';
+
+export const getModelData = async ({ input }: { input: GetByIdInput }) => {
+  try {
+    return await getModel({
+      id: input.id,
+      select: {
+        id: true,
+        name: true,
+        status: true,
+        type: true,
+        uploadType: true,
+        modelVersions: {
+          select: {
+            id: true,
+            name: true,
+            baseModel: true,
+            trainingStatus: true,
+            trainingDetails: true,
+            files: {
+              select: {
+                id: true,
+                url: true,
+                type: true,
+                metadata: true,
+                sizeKB: true,
+                visibility: true,
+              },
+              where: { type: { equals: 'Training Data' } },
+            },
+          },
+        },
+      },
+    });
+  } catch (error) {
+    throw throwDbError(error);
+  }
```

Comment:
Maybe this is a little nitpicky, but would be good to throw the right status error if the model is not found
```suggestion
  try {
    const model = await getModel({
      id: input.id,
      select: {
        id: true,
        name: true,
        status: true,
        type: true,
        uploadType: true,
        modelVersions: {
          select: {
            id: true,
            name: true,
            baseModel: true,
            trainingStatus: true,
            trainingDetails: true,
            files: {
              select: {
                id: true,
                url: true,
                type: true,
                metadata: true,
                sizeKB: true,
                visibility: true,
              },
              where: { type: { equals: 'Training Data' } },
            },
          },
        },
      },
    });
    if (!model) throw throwNotFoundError(`No model with id ${input.id}`);
    
    return model
  } catch (error) {
    if (error instanceof TRPCError) throw error;
    throw throwDbError(error);
  }
```
If you think this is not worth it, then omit my comment 😛

Reply:
thats true, i was actually just copying the pattern used elsewhere that has this generic error throwing